### PR TITLE
Add various components

### PR DESCRIPTION
**Pull request by JMW95**
Add LTC4151 I2C Power Monitor, PCA9502 I2C Port Expander, TYS5040 and SRP5030T inductors.

### CODE REVIEW

File: scripts/build_lib_ic.py [hunk 1]
```diff
@@ -1373,6 +1373,87 @@
...
+                    ("~RESET", 1, "in"),
+                ], [
+                    ("GPIO0", 13, "bidi"),
+                    ("GPIO1", 14, "bidi"),
+                    ("GPIO2", 15, "tri"),
+                    ("GPIO3", 16, "tri"),
+                    ("GPIO4", 18, "tri"),
+                    ("GPIO5", 19, "tri"),
+                    ("GPIO6", 20, "tri"),
+                    ("GPIO7", 21, "tri"),
```
Comment (adamgreig) [inline scripts/build_lib_ic.py]: Should `GPIO0` and `GPIO1` be `bidi` and the rest `tri`?

File: scripts/build_mod_ic.py [hunk 2]
```diff
@@ -401,6 +402,21 @@
...
+        "rows": 4,
+        "pins": 24,
+        "pin_pitch": 0.5,
+        "row_pitch": 4.10,
+        "pad_shape": (0.90, 0.24),
+        "ep_shape": (2.10, 2.10),
+        "ep_mask_shape": (0.45, 0.45, 0.4, 0.4),
+        "ep_paste_shape": (0.45, 0.45, 0.4, 0.4),
+        "chip_shape": (4.3, 4.3),
+        "pin_shape": (0.475, 0.45)
```
Comment (adamgreig) [inline scripts/build_mod_ic.py]: From the datasheet pg20 dimension `b` I think the pin width is between `0.18` and `0.30`, while the length is `0.3` to `0.5` - so this is a bit too big?

File: scripts/build_mod_ic.py [hunk 3]
```diff
@@ -401,6 +402,21 @@
...
+    # IPC-7351B: QFN50P430X430X100-25N
+    "HVQFN24-NXP": {
+        "rows": 4,
+        "pins": 24,
+        "pin_pitch": 0.5,
+        "row_pitch": 4.10,
+        "pad_shape": (0.90, 0.24),
+        "ep_shape": (2.10, 2.10),
+        "ep_mask_shape": (0.45, 0.45, 0.4, 0.4),
+        "ep_paste_shape": (0.45, 0.45, 0.4, 0.4),
```
Comment (adamgreig) [inline scripts/build_mod_ic.py]: I assume the `0.4` is deliberately bigger than the `0.15` the footprint has?
http://www.nxp.com/documents/reflow_soldering/sot616-1_fr.pdf

Fine if that's the case as they do seem quite close together.

File: scripts/build_mod_ic.py (hunk 2 above)
Comment (JMW95) [inline scripts/build_mod_ic.py]: Going by the footprint layout here:
http://www.nxp.com/documents/outline_drawing/SOT616-1.pdf
dimensions D and C are correct as I have them, but that does seem far too large...
I'll change it to 0.5 to match the product datasheet.

File: scripts/build_lib_ic.py (hunk 1 above)
Comment (JMW95) [inline scripts/build_lib_ic.py]: Should all be bidi, will fix.

File: scripts/build_mod_ic.py (hunk 3 above)
Comment (JMW95) [inline scripts/build_mod_ic.py]: Yes - 0.4 was to ensure there were only 2x2 pads generated.

File: scripts/build_mod_ic.py (hunk 2 above)
Comment (adamgreig) [inline scripts/build_mod_ic.py]: Ah, I actually meant the `pin_shape` property (which is just the Fab layer drawing of how wide the pin is). Thinking about it you should have a negative x-dimension too as the pins are internal. If you turn on the F.Fab layer when looking at the result you should see what I mean.

I think the `pad_shape` should continue to match the recommended footprint of CxD.

**Pull request merged**